File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = "";
        
		StringBuilder sb = new StringBuilder(my_string);
		
		for(int i = 0; i< queries.length; i++) {
			int start = queries[i][0];
			int end = queries[i][1];
			
			while(start < end) {
				char c1 = sb.charAt(start);
				char c2 = sb.charAt(end);
				sb.setCharAt(start, c2);
				sb.setCharAt(end, c1);
                
				start++; end--;
			}
		}
        
        answer = sb.toString();
        
        return answer;
    }
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

// queries_rows는 2차원 배열 queries의 행 길이, queries_cols는 2차원 배열 queries의 열 길이입니다.
// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
char* solution(const char* my_string, int** queries, size_t queries_rows, size_t queries_cols) {
    char* answer = (char*)malloc(strlen(my_string) + 1);
    strcpy(answer, my_string);

    for(int i = 0; i < queries_rows; i++){
        int s = queries[i][0];
        int e = queries[i][1];
        
        while (s < e){
            char temp = answer[s];
            answer[s] = answer[e];
            answer[e] = temp;
            s++;
            e--;
        }

    }
    
    
    return answer;
}